source/plugins/go/src/network_flow_logs.go

```diff
@@ -0,0 +1,380 @@
+
+		var dataMap map[string]interface{}
+		var networkFlowLogsMsgPackEntries []NetworkFlowMsgPackEntry
+		numNetworkLogRecords := 0
+
+		for _, record := range tailPluginRecords {
+			networkFlowLogRecordInterface, err := convertFluentBitRecord(record)
+			if err != nil {
+				Log("Error::networkflow:: failed to convert fluent-bit record: %v", err.Error())
+				continue
+			}
+			networkFlowLogRecord, ok := networkFlowLogRecordInterface.(map[string]interface{})
+			if !ok {
+				Log("Error::networkflow:: failed to convert networkflow record to map[string]interface{}: %v", networkFlowLogRecordInterface)
+				continue
+			}
+			dataMap = make(map[string]interface{})
+			if err := mapNetworkFlowLogsToDataMap(dataMap, networkFlowLogRecord); err != nil {
+				Log("Error::networkflow:: failed to map networkflow logs to data map: %v", err.Error())
+				continue
+			}
+			var networkFlowLogsMsgPackEntry NetworkFlowMsgPackEntry
+			networkFlowLogsMsgPackEntry = NetworkFlowMsgPackEntry{
+				Time:   time.Now().Unix(),
+				Record: dataMap,
+			}
+			networkFlowLogsMsgPackEntries = append(networkFlowLogsMsgPackEntries, networkFlowLogsMsgPackEntry)
+		}
+
+		if len(networkFlowLogsMsgPackEntries) > 0 {
+			if IsAADMSIAuthMode == true {
+				MdsdNetworkFlowLogsStreamTagName = getOutputStreamIdTag(RetinaNetworkFlowLogsStreamName, MdsdNetworkFlowLogsStreamTagName, &NetworkFlowTagRefreshTracker)
+				if MdsdNetworkFlowLogsStreamTagName == "" {
+					Log("Error::mdsd::Failed to get stream tag for networkflow logs. Will retry ...")
+					return output.FLB_RETRY
+				}
+			}
+			if MdsdNetworkFlowClient == nil {
+				Log("Error::mdsd::mdsd connection does not exist for networkflow mdsd client. re-connecting ...")
+				CreateMDSDClient(NetworkFlowLogs, ContainerType)
+				if MdsdNetworkFlowClient == nil {
+					Log("Error::mdsd::Unable to create mdsd client for networkflow. Please check error log.")
+					NetworkFlowTelemetryMutex.Lock()
+					defer NetworkFlowTelemetryMutex.Unlock()
+					NetworkFlowLogsMDSDClientCreateErrors += 1
+
+					return output.FLB_RETRY
+				}
+			}
+
+			bts, er := writeNetworkFlowMsgPackEntries(MdsdNetworkFlowClient, MdsdNetworkFlowLogsStreamTagName, networkFlowLogsMsgPackEntries)
+			elapsed = time.Since(start)
+
+			if er != nil {
+				Log("Error::mdsd::Failed to write to mdsd %d records for networkflow logs after %s. Will retry ... error : %s", len(networkFlowLogsMsgPackEntries), elapsed, er.Error())
+				if MdsdNetworkFlowClient != nil {
+					MdsdNetworkFlowClient.Close()
+					MdsdNetworkFlowClient = nil
+				}
+
+				NetworkFlowTelemetryMutex.Lock()
+				defer NetworkFlowTelemetryMutex.Unlock()
+				NetworkFlowLogsMDSDClientCreateErrors += 1
+
+				return output.FLB_RETRY
+			} else {
+				numNetworkLogRecords = len(networkFlowLogsMsgPackEntries)
+				Log("Success::mdsd::Successfully flushed %d networkflow log records that was %d bytes to mdsd in %s ", numNetworkLogRecords, bts, elapsed)
+			}
+		}
+
+		//TODO Telemetry
```

Comment:
We would need to emit both NetworkFlowLogsCount and Size metric similarly containerlogs